CIP-0PRS/README.lagda.md

```diff
@@ -0,0 +1,1604 @@
+
+| Provider     |     VM | 50 kB/s | 125 kB/s | 250 kB/s |
+| ------------ | -----: | ------: | -------: | -------: |
+| DigitalOcean |   $188 |    $188 |     $188 |     $188 |
+| Google Cloud |   $200 |    $214 |     $234 |     $268 |
+| AWS          | ? $150 |    $161 |     $178 |     $206 |
+| Azure        |   $175 |    $186 |     $202 |     $230 |
+| OVH          |    $70 |     $70 |      $70 |      $70 |
+| Hetzner      |    $32 |     $32 |      $32 |      $32 |
+
+[^9]: https://cexplorer.io/usage
+
+[^10]: https://iohk.io/en/research/library/papers/mind-your-outcomes-the-dqsd-paradigm-for-quality-centric-systems-development-and-its-application-to-a-blockchain-case-study/
+
+[^11]: https://pooltool.io/networkhealth
+
+[^12]: https://developers.cardano.org/docs/operate-a-stake-pool/hardware-requirements/
+
+[^13]: The AWS cost is quite hard to estimate up-front. The $150 base price is a rough average of various instances options in the target range. Google, AWS and Azure prices are based on 100% uptime and at least 1-year reservation for discounts. Cloud providers only charge _outgoing_ network traffic. The actual cost per GB depends on the destination, this table assumes all outbound traffic will be targeted outside of the provider which obviously won't be true, so it should be treated as an upper bound.
+
+#### Persistent storage
+
+Under similar assumptions, we can estimate the storage requirements entailed by Peras: ignoring the impact of cool-down periods, which last for a period at least as long as $k$ blocks, the requirement to store certificates for every round increases node's storage by about **20%**. Votes are expected to be kept in memory so their impact on storage will be null.
+
+#### CPU
+
+The [Peras Technical Report #2](https://peras.cardano-scaling.org/docs/reports/tech-report-2#votes--certificates) provides some models and benchmarks for votes generation, votes verification, certificates proving and certificates verification, and votes diffusion. Those benchmarks are based on efficient sortition-based voting and ALBAs certificate, and demonstrate the impact of Peras on computational resources for a node will be minimal. Moreover, the most recent version of the algorithm detailed in this report is designed in such a way the voting process runs in parallel with block production and diffusion and therefore is not on this critical path. For ALBA certificates, assuming 1000 votes, a honest to faulty ratio of 80/20, and security parameter $λ=128$, one has the following typical measurements.
```

Comment:
Could we clarify a bit more the details of the parameters of Alba (as described in the [original paper](https://eprint.iacr.org/2023/1655.pdf)) in order to obtain the certificate size of 47Kb. After reading the report above, I am having some difficulty mapping the parameters from the report to the parameters of the paper? 

Additionally, the estimated size of 47Kb is a bit unclear to me - we talk about honest / faulty ratio of 80/20 and (what seems to be assumed) is that $n_f$ is set to 20% and not the Peras quorum of 75% to obtain this value of 47Kb (I might be wrong here, so correct me if this is the case). This 20% coincides with the adversarial votes and hence, I am asking, what statement exactly is the certificate endorsing? Relating to the Alba abstract problem in the paper, Is it simply saying that "I know a set of voters and at least one of them is honest" ? If so, why is this sufficient for the consensus / boosting the block? I’d appreciate some clarifications on that.

Reply:
The 47kB value might have come from an earlier version of ALBA prototype in Haskell, the latest value I computed (git hash 08cb09accf47093f53cf55e9b344f15cc9a3a408) gives me

```
 % cabal run alba -- prove --len 710
Generating random items
Generating proof Options {size = 100, len = 710, params = Params {λ_sec = 128, λ_rel = 128, n_p = 80, n_f = 20}, input = Nothing, output = "proof.alba"}
Written proof to 'proof.alba' (50278 bytes, 0 retries)
```

The parameters 80/20 used the computation came from my misunderstanding of what's actually guaranteed by ALBA certificates as outlined in [this discussion](https://github.com/cardano-scaling/alba/discussions/17) and as noted in one of the comments, this needs to be fixed to account for more serious assessment of security implication of ALBA parameters on Peras.
This is the purpose of https://github.com/cardano-scaling/alba/pull/22 and the discussion we've had on Friday. This CIP and the report will be corrected to tone down the enthusiasm and expectations around ALBA as it's perfectly possible we realise these are not suitable for Peras after all.

Reply:
I'm resolving this because the ALBA-specific discussion was moved to the Voting CIP.